pkg/kubectl/cmd/edit.go

```diff
@@ -156,14 +169,18 @@ func RunEdit(f *cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args
 		// compare two different GroupVersions).
 		Decoder: f.Decoder(false),
 	}
-
-	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
-		NamespaceParam(cmdNamespace).DefaultNamespace().
+	var b *resource.Builder
+	if isUsedByEditCmd {
+		b = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+			ResourceTypeOrNameArgs(true, args...).
+			Latest()
+	} else {
+		b = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), runtime.UnstructuredJSONScheme)
```

Comment:
This is because I want to support the use case:
`kubectl create -f invalid-k8s-resource.yaml --edit`
where 'invalid-k8s-resource.yaml' contains some invalid configuration. User can edit these invalid fields before creating them. I think it will make `--edit` more useful.
Otherwise, it will simply return an error, the editor won't even pop out to let user edit it.